subsys/bluetooth/services/fast_pair/adv_manager/locator_tag.c

```diff
@@ -0,0 +1,210 @@
+		return;
+	}
+
+	bt_fast_pair_adv_manager_request(&bt_fast_pair_adv_manager_trigger_clock_sync, enable);
+}
+
+static void fp_adv_clock_synced(void)
+{
+	LOG_DBG("Fast Pair Adv Manager: Locator Tag: clock information synchronized with "
+		"the authenticated Bluetooth peer");
+
+	if (fmdn_provisioned) {
+		/* Fast Pair Implementation Guidelines for the locator tag use case:
+		 * After a power loss, the device should advertise non-discoverable
+		 * Fast Pair frames until the next invocation of read beacon parameters.
+		 * This lets the Seeker detect the device and synchronize the clock even
+		 * if a significant clock drift occurred.
+		 */
+		fp_adv_clock_sync_trigger_manage(false);
+	}
+}
+
+static void fp_adv_provisioning_state_changed(bool provisioned)
+{
+	fmdn_provisioned = provisioned;
+
+	if (!bt_fast_pair_adv_manager_is_ready()) {
+		return;
+	}
+
+	if (provisioned) {
+		fp_adv_fmdn_provisioning_trigger_manage(false);
+	} else {
+		fp_adv_clock_sync_trigger_manage(false);
+	}
+}
+
+static struct bt_fast_pair_fmdn_info_cb fmdn_info_cb = {
+	.clock_synced = fp_adv_clock_synced,
+	.provisioning_state_changed = fp_adv_provisioning_state_changed,
+};
+
+static void fp_adv_account_key_written(struct bt_conn *conn)
+{
+	/* The first and only Account Key write starts the FMDN provisioning. */
+	if (!fmdn_provisioned && !fp_account_key_present &&
+	    bt_fast_pair_adv_manager_is_ready()) {
+		fp_adv_fmdn_provisioning_trigger_manage(true);
+	}
+
+	fp_account_key_present = bt_fast_pair_has_account_key();
+}
+
+static struct bt_fast_pair_info_cb fp_info_callbacks = {
+	.account_key_written = fp_adv_account_key_written,
+};
+
+static bool fp_adv_is_first_enable_after_bootup(void)
+{
+	static bool first_enable_after_bootup = true;
+	bool is_first_enable_after_bootup = first_enable_after_bootup;
+
+	first_enable_after_bootup = false;
+
+	return is_first_enable_after_bootup;
+}
+
+int fp_adv_manager_locator_tag_enable(void)
+{
+	bool is_first_enable_after_bootup = fp_adv_is_first_enable_after_bootup();
+
+	fp_account_key_present = bt_fast_pair_has_account_key();
+	fmdn_provisioned = bt_fast_pair_fmdn_is_provisioned();
+
+	/* Handle the clock synchronization trigger. */
+	if (is_first_enable_after_bootup && fmdn_provisioned) {
+		fp_adv_clock_sync_trigger_manage(true);
+	} else if (!fmdn_provisioned) {
+		fp_adv_clock_sync_trigger_manage(false);
+	} else {
+		/* Left empty intentionally: this state is used to recreate the trigger state
+		 * after the bt_fast_pair_adv_manager_disable operation.
+		 */
+	}
+
+	/* Handle the FMDN provisioning trigger. */
+	if (fp_account_key_present && !fmdn_provisioned && !is_first_enable_after_bootup) {
+		/* The FMDN provisioning trigger is inactive as the Fast Pair Advertising Manager
+		 * gets disabled (bt_fast_pair_adv_manager_disable) by the user during the
+		 * provisioning procedure.
+		 */
+		LOG_WRN("Fast Pair Adv Manager: Locator Tag: FMDN provisioning trigger inactive");
+	}
```

Comment:
Could you maybe rewrite this logic a bit and do:
```
if (!is_first_enable_after_bootup) {
    // do whatever is needed
    return 0;
}
// do everything else
```
That would be way easier to understand.